File: frontend/src/types/weeklyDashboard.ts
export interface WeeklyDashboardItem {
    category_id: number
    category_name: string

    planned_minutes: number
    actual_minutes: number
    remaining_minutes: number
    progress_percent: number
}


export interface WeeklyDashboardDay {
    date: string
    day_of_week: number
    day_name: string

    items: WeeklyDashboardItem[]
}


export interface WeeklyDashboard {
    week_start: string
    week_end: string

    days: WeeklyDashboardDay[]
}

File: backend/app/schemas/weekly_dashboard.py
from datetime import date

from pydantic import BaseModel


class WeeklyDashboardItem(BaseModel):
    category_id: int
    category_name: str

    planned_minutes: int
    actual_minutes: int
    remaining_minutes: int
    progress_percent: float


class WeeklyDashboardDay(BaseModel):
    date: date
    day_of_week: int
    day_name: str

    items: list[WeeklyDashboardItem]


class WeeklyDashboardResponse(BaseModel):
    week_start: date
    week_end: date

    days: list[WeeklyDashboardDay]